go/interpreter/lfvm/instructions_test.go

```diff
@@ -1153,49 +1153,86 @@ func TestComputeCodeSizeCost(t *testing.T) {
 	}
 }
 
-func TestGenericCreate_MaxInitCodeSizeIsNotCheckedBeforeShanghai(t *testing.T) {
-
-	tests := []struct {
+func TestGenericCreate_ReportsErrors(t *testing.T) {
+	one := uint256.NewInt(1)
+	tests := map[string]struct {
+		offset, size  uint256.Int
+		kind          tosca.CallKind
 		revision      tosca.Revision
 		expectedError error
 	}{
-		{tosca.R11_Paris, nil},
-		{tosca.R12_Shanghai, errInitCodeTooLarge},
+		"not enough gas for code size": {
+			offset:        *one,
+			size:          *uint256.NewInt(31),
+			revision:      tosca.R12_Shanghai,
+			expectedError: errOutOfGas,
+		},
+		"gas not checked for max code size before shanghai": {
+			offset:        *one,
+			size:          *uint256.NewInt(31),
+			revision:      tosca.R11_Paris,
+			expectedError: nil,
```

Comment:
nit: maybe `kind` should be specified in both cases since the default value for `tosca.CallKind` is actually `Call`, not `Create`.